e2e/test/iothub/service/DigitalTwinClientE2ETests.cs

```diff
@@ -0,0 +1,203 @@
+using Microsoft.Azure.Devices.Serialization;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
+using PnpHelpers;
+
+namespace Microsoft.Azure.Devices.E2ETests.Iothub.Service
+{
+    [TestClass]
+    [TestCategory("E2E")]
+    [TestCategory("IoTHub")]
+    public class DigitalTwinClientE2ETests : E2EMsTestBase
+    {
+        private const string ThermostatModelId = "dtmi:com:example:Thermostat;1";
+        private const string TemperatureControllerModelId = "dtmi:com:example:TemperatureController;1";
+
+        private readonly string _devicePrefix = $"E2E_{nameof(DigitalTwinClientE2ETests)}_";
+        private static readonly string s_connectionString = Configuration.IoTHub.ConnectionString;
+
+        [LoggedTestMethod]
+        public async Task DigitalTwinWithOnlyRootComponentOperationsAsync()
+        {
+            // Create a new test device instance.
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, _devicePrefix).ConfigureAwait(false);
+            string deviceId = testDevice.Id;
+
+            try
+            {
+                // Create a device client instance over Mqtt, initializing it with the "Thermostat" model which has only a root component.
+                var options = new ClientOptions
+                {
+                    ModelId = ThermostatModelId,
+                };
+                using DeviceClient deviceClient = testDevice.CreateDeviceClient(Client.TransportType.Mqtt, options);
+
+                // Call openAsync() to open the device's connection, so that the ModelId is sent over Mqtt CONNECT packet.
+                await deviceClient.OpenAsync().ConfigureAwait(false);
+
+                // Perform operations on the digital twin.
+                using var digitalTwinClient = DigitalTwinClient.CreateFromConnectionString(s_connectionString);
+
+                // Retrieve the digital twin.
+                Rest.HttpOperationResponse<ThermostatTwin, DigitalTwinGetHeaders> response =
+                    await digitalTwinClient.GetAsync<ThermostatTwin>(deviceId).ConfigureAwait(false);
+                ThermostatTwin twin = response.Body;
+                twin.Metadata.ModelId.Should().Be(ThermostatModelId);
+
+                // Set callback handler for receiving root-level twin property updates.
+                await deviceClient.SetDesiredPropertyUpdateCallbackAsync((patch, context) =>
+                {
+                    Logger.Trace($"{nameof(DigitalTwinWithComponentOperationsAsync)}: DesiredProperty update received: {patch}, {context}");
+                    return Task.FromResult(true);
+                }, deviceClient);
+
+                // Update the root-level property "targetTemperature".
+                string propertyName = "targetTemperature";
+                double propertyValue = new Random().Next(0, 100);
+                var ops = new UpdateOperationsUtility();
+                ops.AppendAddOp($"/{propertyName}", propertyValue);
```

Comment:
Does it work without the root slash?

Reply:
not sure, let me try it out

Reply:
No, it doesn't. I can create a helper for this as well.

`Message=ErrorCode:InvalidDigitalTwinPatch;Error in path: targetTemperature. 'path' string must be non-null, begin with '/' and have atleast the component name.`

Reply:
We should document it in the very least. A helper might be an overkill. I'd almost rather the client code look for a slash at the beginning, and prepend one. I'm curious what the Azure SDK team would suggest.

Reply:
the client code is will get the serialized patch, so for it to inspect it and append a "/" if missing, we would have to deserialize and serialize it again. 
I can add it to the update utility, to always append a "/", but that might complicate it when we have property within a component (we might need another overload).

A helper might be the simplest option here.

Reply:
Discussed offline. Also, this is pending recommendation from sdk board.